File: compiler/bench/binsplit.fs
\ expected: 1048576
\ Binary split recursion, depth 20, 1000 times

: binsplit ( depth -- count ) recursive
  dup 0= if drop 1 exit then
  1- dup recurse swap recurse + ;

: main
  1000 0 do 20 binsplit drop loop
  20 binsplit . cr ;

File: compiler/bench/binsplit.c
// expected: 1048576
#include <stdio.h>
long binsplit(long depth) {
    if (depth == 0) return 1;
    return binsplit(depth - 1) + binsplit(depth - 1);
}

int main() {
    for (long i = 0; i < 1000; i++) {
        binsplit(20);
    }
    printf("%ld\n", binsplit(20));
    return 0;
}
